### DISCUSSION TITLE
Loading cookies from a local cookie file

### DISCUSSION THREAD

**Discussion by achyudh** [qutebrowser/qutebrowser#8259] [General] [answered]
I have a local cookie file in Netscape cookie file format which I would like to load, either on startup or using a command. These are cookies that I need to access internal websites at work and for development/testing. What would be the best way to go about this in qutebrowser? I currently achieve this in Firefox using a WebExtension.

I was thinking maybe it would be possible to use [QWebEngineCookieStore](https://doc.qt.io/qt-6/qwebenginecookiestore.html) in an external Python script to add the cookies to the store. Totally open to hacky solutions or contributing to qutebrowser if this is not currently possible.

Relevant issue (maybe): https://github.com/qutebrowser/qutebrowser/issues/5500

**Comment by The-Compiler** [comment, marked as answer]
I believe you could add them manually in the `:devtools`, or find a way to insert them into the underlying Chromium cookie sqlite database. Indeed #5500 would be relevant and allow for this, but from what I saw last time I looked at it, I'm not sure it it's a desirable thing to have. It certainly comes with a lot of caveats, and is not something I'd recommend as a contribution for someone (I'm assuming, correct me if I'm wrong) not familiar with the qutebrowser codebase and QtWebEngine.

**Comment by achyudh** [comment]
Ah I see, I first considered using a script that inserts the cookies into the database but it looks like Chromium locks the database while it is running #6680. I ended up using [mitmproxy](https://docs.mitmproxy.org/stable/overview-features/) and while not ideal, it works quite ideal for my use case.
